### PR TITLE
Clean build configurations

### PR DESCRIPTION
**Pull request by Odrakir**
This is what (I think) we need:

ApiTests
Debug
UnitTests
UITests
Debug.Daneel
Debug.Giskard
Release.Daneel
Release.Giskard
AdHoc.Giskard
AdHoc.Daneel

No platform (iPhone or iPhoneSimulator) as the only use for that is to change more specific stuff within the configs. I believe we were not using that. If we were, then this PR is not valid.

Also, please, take a look at what assemblies are selected for each config.

And also the cake script should be reviewed.

Finally, if everything is ok we should also do the necessary changes in Bitrise.

### CODE REVIEW

**Comment by amulware** [comment]
I'd like reviews from people trying it on both android and ios (please specify when reviewing).

If the code looks ok and works for development, then I'll make the changes on bitrise and test all the builds there manually to make sure.

**Comment by Odrakir** [comment]
If possible I would suggest also taking a look at the diff itself, there might be some obscure parameter I deleted accidentally.

**Comment by amulware** [comment]
Yes, I'll definitely do so (and would like a second opinion).

File: Toggl.Daneel.Tests.UI/Toggl.Daneel.Tests.UI.csproj [hunk 1]
```diff
@@ -19,8 +19,10 @@
     <WarningLevel>4</WarningLevel>
     <CodeAnalysisRuleSet>..\Toggl.ruleset</CodeAnalysisRuleSet>
   </PropertyGroup>
-  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Release|AnyCPU' ">
-    <OutputPath>..\bin\Release</OutputPath>
+  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'AdHoc.Daneel|AnyCPU' ">
+    <Optimize>false</Optimize>
+    <OutputPath>bin\AdHoc.Daneel</OutputPath>
```
Comment (willsb) [inline Toggl.Daneel.Tests.UI/Toggl.Daneel.Tests.UI.csproj]: Shouldn't this be `..\bin\AdHoc.Daneel` instead? Also, was the `AdHoc.Daneel` folder put there intentionally?

File: Toggl.Daneel.Tests.UI/Toggl.Daneel.Tests.UI.csproj [hunk 2]
```diff
@@ -19,8 +19,10 @@
...
+    <WarningLevel>4</WarningLevel>
```
Comment (willsb) [inline Toggl.Daneel.Tests.UI/Toggl.Daneel.Tests.UI.csproj]: Could be removed

File: Toggl.Daneel/Toggl.Daneel.csproj [hunk 3]
```diff
@@ -11,49 +11,23 @@
...
-    <DebugSymbols>true</DebugSymbols>
-    <DebugType>full</DebugType>
+  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Release.UITest|iPhoneSimulator' ">
     <Optimize>false</Optimize>
-    <OutputPath>..\bin\iPhoneSimulator\Debug</OutputPath>
-    <DefineConstants>DEBUG;ENABLE_TEST_CLOUD;</DefineConstants>
-    <ErrorReport>prompt</ErrorReport>
+    <OutputPath>bin\iPhoneSimulator\Release.UITest</OutputPath>
+    <DefineConstants>
+    </DefineConstants>
```
Comment (willsb) [inline Toggl.Daneel/Toggl.Daneel.csproj]: Those constants are needed for UITests

File: Toggl.Daneel/Toggl.Daneel.csproj [hunk 4]
```diff
@@ -11,49 +11,23 @@
...
   </PropertyGroup>
-  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Debug|iPhoneSimulator' ">
-    <DebugSymbols>true</DebugSymbols>
-    <DebugType>full</DebugType>
+  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Release.UITest|iPhoneSimulator' ">
     <Optimize>false</Optimize>
-    <OutputPath>..\bin\iPhoneSimulator\Debug</OutputPath>
-    <DefineConstants>DEBUG;ENABLE_TEST_CLOUD;</DefineConstants>
-    <ErrorReport>prompt</ErrorReport>
+    <OutputPath>bin\iPhoneSimulator\Release.UITest</OutputPath>
```
Comment (willsb) [inline Toggl.Daneel/Toggl.Daneel.csproj]: Why remove the `..\` from the release output path?

File: Toggl.Daneel/Toggl.Daneel.csproj [hunk 5]
```diff
@@ -65,41 +39,26 @@
...
-    <MtouchArch>x86_64</MtouchArch>
+    <MtouchLink>SdkOnly</MtouchLink>
     <MtouchHttpClientHandler>HttpClientHandler</MtouchHttpClientHandler>
   </PropertyGroup>
+  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Debug|iPhoneSimulator' ">
+    <OutputPath>bin\iPhoneSimulator\Debug\</OutputPath>
+    <PlatformTarget>iPhoneSimulator</PlatformTarget>
+  </PropertyGroup>
+  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Release.Daneel|AnyCPU' ">
+    <OutputPath>bin\Release.Daneel\</OutputPath>
```
Comment (willsb) [inline Toggl.Daneel/Toggl.Daneel.csproj]: `..\`

File: Toggl.Foundation.MvvmCross/Toggl.Foundation.MvvmCross.csproj [hunk 6]
```diff
@@ -2,15 +2,26 @@
...
   </PropertyGroup>
-  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Release|AnyCPU' ">
-    <OutputPath>..\bin\Release\</OutputPath>
+  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'AdHoc.Daneel|AnyCPU' ">
+    <IntermediateOutputPath>obj\AdHoc.Daneel</IntermediateOutputPath>
+    <DebugType>
+    </DebugType>
+    <Optimize>false</Optimize>
+    <OutputPath>bin\AdHoc.Daneel</OutputPath>
+    <DefineConstants>
```
Comment (willsb) [inline Toggl.Foundation.MvvmCross/Toggl.Foundation.MvvmCross.csproj]: We need some constants for the AdHoc release

File: Toggl.Foundation.Tests/Toggl.Foundation.Tests.csproj [hunk 7]
```diff
@@ -2,15 +2,27 @@
...
   </PropertyGroup>
   <PropertyGroup>
     <CodeAnalysisRuleSet>$(SolutionDir)Toggl.ruleset</CodeAnalysisRuleSet>
   </PropertyGroup>
   <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Debug|AnyCPU' ">
     <OutputPath>..\bin\Debug\</OutputPath>
   </PropertyGroup>
-  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Release|AnyCPU' ">
-    <OutputPath>..\bin\Release\</OutputPath>
+  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'AdHoc.Daneel|AnyCPU' ">
```
Comment (willsb) [inline Toggl.Foundation.Tests/Toggl.Foundation.Tests.csproj]: Why do Foundation tests need to know anything about Daneel?

File: Toggl.Foundation/Toggl.Foundation.csproj [hunk 8]
```diff
@@ -1,15 +1,26 @@
...
   </PropertyGroup>
   <PropertyGroup>
     <CodeAnalysisRuleSet>$(SolutionDir)Toggl.ruleset</CodeAnalysisRuleSet>
   </PropertyGroup>
   <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Debug|AnyCPU' ">
     <OutputPath>..\bin\Debug\</OutputPath>
   </PropertyGroup>
-  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Release|AnyCPU' ">
-    <OutputPath>..\bin\Release\</OutputPath>
+  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'AdHoc.Daneel|AnyCPU' ">
```
Comment (willsb) [inline Toggl.Foundation/Toggl.Foundation.csproj]: Same

File: Toggl.Giskard.Tests/Toggl.Giskard.Tests.csproj [hunk 9]
```diff
@@ -18,6 +18,7 @@
     <MonoAndroidResourcePrefix>Resources</MonoAndroidResourcePrefix>
     <MonoAndroidAssetsPrefix>Assets</MonoAndroidAssetsPrefix>
     <AndroidUseLatestPlatformSdk>true</AndroidUseLatestPlatformSdk>
+    <AndroidLinkMode>None</AndroidLinkMode>
```
Comment (willsb) [inline Toggl.Giskard.Tests/Toggl.Giskard.Tests.csproj]: Do Link, please.

File: Toggl.Giskard/Toggl.Giskard.csproj [hunk 10]
```diff
@@ -34,34 +34,16 @@
...
-    </AndroidLinkSkip>
+    <AndroidLinkSkip></AndroidLinkSkip>
     <AndroidLinkMode>SdkOnly</AndroidLinkMode>
-    <AndroidSupportedAbis>armeabi-v7a;armeabi;arm64-v8a;x86_64</AndroidSupportedAbis>
-    <AndroidHttpClientHandlerType>Xamarin.Android.Net.AndroidClientHandler</AndroidHttpClientHandlerType>
-    <EmbedAssembliesIntoApk>true</EmbedAssembliesIntoApk>
     <AndroidSupportedAbis>armeabi-v7a;armeabi;x86_64;x86;arm64-v8a</AndroidSupportedAbis>
+    <AndroidHttpClientHandlerType>Xamarin.Android.Net.AndroidClientHandler</AndroidHttpClientHandlerType>
     <DebugType></DebugType>
-    <DefineConstants>USE_ANALYTICS;USE_PRODUCTION_API</DefineConstants>
```
Comment (willsb) [inline Toggl.Giskard/Toggl.Giskard.csproj]: y tho?

**Comment by Odrakir** [comment]
I'm doing another PR with this... Yes, I know...

**Pull request closed without merging**